### DISCUSSION TITLE
ubuntu base image & hardware acceleration

### DISCUSSION THREAD

**Discussion by kaytwo** [immich-app/immich#1401] [Q&A]
Hi! Fantastic project! I was wondering if there is context for the attempt and reversion of using ubuntu as the base image: (#851 and #863)?

The reason I ask is that I'm looking into using hardware acceleration for the transcoding and machine learning pipelines. Inspired by [Frigate NVR](https://docs.frigate.video/configuration/hardware_acceleration/), it looks feasible to enable both hardware accelerated video encoding (via ffmpeg with the correct hardware libraries), as well as speeding up the ML pipelines by using tflite with something like a [Google Coral](https://github.com/tensorflow/sig-tfjs/tree/main/tfjs-tflite-node-codelab/coral_inference_working) or CUDA via [tfjs-node-gpu](https://github.com/tensorflow/tfjs/blob/master/tfjs-node/README.md).

The first step to a lot of these improvements would be to transition to ubuntu so that the libraries needed for interfacing with these hardware accelerators will be more easily used, either via ffmpeg or tensorflow. Would love to hear any feedback about how desirable or feasible this would be.

**Comment by bo0tzz** [comment]
I don't remember with certainty, but I believe we reverted it because it had some issue with one of our dependencies in the ubuntu arm64 base image. It also (recently) turned out that the issue we were trying to solve by moving to an ubuntu image (namely, DNS resolution in Kubernetes clusters) can be solved another way as well.

I believe @zackpollard was intending to look into ffmpeg hwaccel at some point (Also see #203), so he'll know more about that topic than I do. As for the ML system, the current container is intended to be a placeholder so I don't know if it's worthwhile to put energy into that just yet.

**Reply by dpnvektor** [reply]
Issue [1225](https://github.com/immich-app/immich/issues/1225) addresses the hardware acceleration question.
